library/src/main/java/com/otaliastudios/zoom/ZoomLayout.kt

```diff
@@ -80,6 +80,24 @@ private constructor(context: Context, attrs: AttributeSet?, @AttrRes defStyleAtt
 
     //region Internal
 
+    private class LayoutListener constructor(engine: ZoomEngine, childView: View) : ViewTreeObserver.OnGlobalLayoutListener {
+
+        private val mEngine = WeakReference(engine)
+        private val mChildView = WeakReference(childView)
+
+        override fun onGlobalLayout() {
+
+            val engine = mEngine.get()
+            val childView = mChildView.get()
+
+            if (childView == null || engine == null) {
+                return
```

Comment:
When the `childView` is not `null` the listener should be able to remove itself, right?

Reply:
My first proposal in the discussion of #58 was indeed to have the listener self-remove the first time it is fired (provided that `childView` is not `null`), but @natario1 pointed out that such behavior would not be correct. I then implemented the `WeakReference` approach so as to avoid leaking the view in case the listener is not removed by the system.

Reply:
@dmazzoni I am not talking about removing it the first time but when the engine is null and the view is not. But a simple `return` doesn't remove the listener ever - except when the view is destroyed I guess. Still seems like a workaround, what about @natario1 ?

Reply:
@dmazzoni I would relly like to merge this PR but the current solution does not seem optimal and I don't know how to reproduce this issue so it would be pretty cool if you could test if detaching it in `ZoomLayout.onDetachedFromWindow()` as @natario1 suggested also fixes the issue. If not I would merge the current solution (if @natario1 approves) to get this resolved. We can always improve later.
Thx for your work and interest!